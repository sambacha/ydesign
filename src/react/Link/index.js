import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { system } from 'styled-system';
import Box, { StyledBox } from '../Box';
import theme from '../theme';

const hoverColor = system({
  hoverColor: {
    property: 'color',
    scale: 'colors',
  },
});

const activeColor = system({
  activeColor: {
    property: 'color',
    scale: 'colors',
  },
});

const StyledLink = styled(StyledBox)`
  & {
    text-decoration: none;
    cursor: pointer;
  }

  &:hover {
    text-decoration: underline;
    ${hoverColor};
  }

  &:active {
    text-decoration: none;
    ${activeColor};
  }
`;

const Link = React.forwardRef((props, ref) => (
  <StyledLink as={'a'} ref={ref} {...props} />
));

Link.defaultProps = {
  color: 'primary',
  hoverColor: 'primary-light',
  activeColor: 'primary-dark',
  fontSize: 1,
  fontWeight: 3,
};

Link.propTypes = {
  ...Box.PropTypes,
  theme: PropTypes.object,
};

Link.displayName = 'Link';

export default Link;
