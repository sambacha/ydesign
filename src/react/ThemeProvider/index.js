import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';
import theme from '../theme';

const ThemeProvider = ({ children, theme, ...rest }) => {
  return (
    <StyledThemeProvider theme={theme} {...rest}>
      {children}
    </StyledThemeProvider>
  );
};

ThemeProvider.propTypes = {
  theme: PropTypes.object,
};

ThemeProvider.defaultProps = {
  theme,
};

ThemeProvider.displayName = 'ThemeProvider';

export default ThemeProvider;
