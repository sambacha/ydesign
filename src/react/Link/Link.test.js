import Link from './';
import ThemeProvider from '../ThemeProvider';
import theme from '../theme';

describe('Link component sanity', () => {
  it('has name', () => {
    expect(Link.displayName).toBe('Link');
  });

  it('matches default snapshot', () => {
    const component = render(<Link />);
    expect(component).toMatchSnapshot();
  });
  it('matches themed snapshot', () => {
    const component = render(
      <ThemeProvider theme={theme}>
        <Link />
      </ThemeProvider>
    );
    expect(component).toMatchSnapshot();
  });
});
