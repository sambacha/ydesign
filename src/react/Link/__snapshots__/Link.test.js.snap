// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component sanity matches default snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  color: #4E3FCE;
  font-size: 14px;
  font-weight: 600;
}

.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #7165d7;
}

.c0:active {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #3e32a4;
}

<a
  class="sc-AxjAm c0"
  color="primary"
  font-size="1"
  font-weight="3"
/>
`;

exports[`Link component sanity matches themed snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  color: #4E3FCE;
  font-size: 14px;
  font-weight: 600;
}

.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #7165d7;
}

.c0:active {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #3e32a4;
}

<a
  class="sc-AxjAm c0"
  color="primary"
  font-size="1"
  font-weight="3"
/>
`;
